test/stream/bridge/keboola/keboola_test.go

```diff
@@ -0,0 +1,433 @@
+package keboola_test
+
+import (
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/c2h5oh/datasize"
+	"github.com/keboola/go-client/pkg/keboola"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/encoding/json"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/duration"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/servicectx"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/config"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/diskwriter/network"
+	stagingConfig "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/staging/config"
+	targetConfig "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/target/config"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
+)
+
+type withProcess interface {
+	Process() *servicectx.Process
+}
+
+func TestKeboolaBridgeWorkflow(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
+	defer cancel()
+
+	// Update configuration to make the cluster testable
+	configFn := func(cfg *config.Config) {
+		// Use deterministic load balancer
+		cfg.Storage.Level.Local.Writer.Network.PipelineBalancer = network.RoundRobinBalancerType
+
+		// In the test, we trigger the slice upload via the records count, the other values are intentionally high.
+		cfg.Storage.Level.Staging.Upload.Trigger = stagingConfig.UploadTrigger{
+			Count:    10,
+			Size:     1000 * datasize.MB,
+			Interval: duration.From(30 * time.Minute),
+		}
+
+		// In the test, we trigger the file import via the records count, the other values are intentionally high.
+		cfg.Storage.Level.Target.Import.Trigger = targetConfig.ImportTrigger{
+			Count:       30,
+			Size:        1000 * datasize.MB,
+			Interval:    duration.From(30 * time.Minute),
+			SlicesCount: 100,
+			Expiration:  duration.From(30 * time.Minute),
+		}
+	}
+
+	ts := setup(t, ctx, configFn)
+	defer ts.teardown(t, ctx)
+
+	ts.testSlicesUpload(
+		t,
+		ctx,
+		0,
+		20,
+		[]model.File{
+			{State: model.FileWriting},
+		},
+		[]model.Slice{
+			{State: model.SliceWriting},
+			{State: model.SliceWriting},
+		},
+		[]model.Slice{
+			{State: model.SliceUploaded},
+			{State: model.SliceWriting},
+			{State: model.SliceUploaded},
+			{State: model.SliceWriting},
+		},
```

Comment:
Some structure may be better, I don't know at first look, what the numbers mean.

Reply:
I know, could not think a better solution than this atm. Need to think it through more, but don't want to spend too much time for overengineering :sweat_smile:

Reply:
https://github.com/keboola/keboola-as-code/pull/1983/commits/62321d71e871d3b0610c47833a0dc94ba57078ef